src/groupeddataframe/splitapplycombine.jl

```diff
@@ -837,22 +910,12 @@ function (agg::Aggregate{typeof(length)})(incol::AbstractVector, gd::GroupedData
     end
 end
 
-function do_f(f, x...)
-    @inline function fun(x...)
-        res = f(x...)
-        if res isa Union{AbstractDataFrame, NamedTuple, DataFrameRow, AbstractMatrix}
-            throw(ArgumentError("a single value or vector result is required when passing " *
-                                "multiple functions (got $(typeof(res)))"))
-        end
-        res
-    end
-end
-
 isagg(p::Pair) = check_aggregate(last(p)) isa AbstractAggregate && first(p) isa ColumnIndex
 
 const MULTI_COLS_TYPE = Union{AbstractDataFrame, NamedTuple, DataFrameRow, AbstractMatrix}
 
 function _combine(f::AbstractVector{<:Pair}, gd::GroupedDataFrame, nms::AbstractVector{Symbol})
+    # here f should be normalized and in a form of source_cols => fun
```

Comment:
Is "here" the function, or just the first branch? Why not put this in the signature to enforce it?

Reply:
"here" means all elements of vector `f`. I will add a signature to enforce it.